packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -1,169 +1,489 @@
+          setActiveValues((prev) => prev.slice(0, prev.length - 1));
+        }
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  return (
+    <ComboboxContext.Provider
+      value={{
+        size,
+        values,
+        activeValues,
+        multiple,
+        showEmptyChild,
+        activeIndex,
+        disabled,
+        readOnly,
+        label,
+        description,
+        hideLabel,
+        open,
+        inputRef,
+        refs,
+        inputValue,
+        activeDescendant,
+        error,
+        formFieldProps,
+        name,
+        setInputValue,
+        setActiveIndex,
+        handleKeyDown,
+        setOpen,
+        getReferenceProps,
+        setActiveValues,
+        /* Recieves index of item, and the ID of the button element */
+        setActiveItem: (index: number, id: string) => {
+          if (readOnly) return;
+          if (disabled) return;
+          setActiveIndex(index);
+          setActiveDescendant(id);
+        },
+        /* Recieves the value of the item, and searches for it in our values lookup */
+        onItemClick: (value: string) => {
+          if (readOnly) return;
+          if (disabled) return;
+          let item;
+          for (const valueItem of values) {
+            if (valueItem.value === value) {
+              item = valueItem;
+              break;
+            }
+          }
+          handleSelectItem(item as ValueItemType);
+        },
+      }}
+    >
+      <Box
+        className={classes.combobox}
+        ref={portalRef}
+      >
+        {/* This is only for the Combobox to work in forms */}
+        {name && <ComboboxNative />}
+
+        <ComboboxLabel />
+        <ComboboxInput />
+        {formFieldProps.hasError && <ComboboxError />}
+      </Box>
+
+      {/* This is the floating list with items */}
+      {open && (
+        <FloatingPortal root={portal ? null : portalRef}>
+          <FloatingFocusManager
+            context={context}
+            initialFocus={-1}
+            visuallyHiddenDismiss
+          >
+            <Box
+              shadow='medium'
+              borderRadius='medium'
+              aria-labelledby={formFieldProps.inputProps.id}
+              aria-autocomplete='list'
+              {...getFloatingProps({
+                ref: refs.setFloating,
+                style: {
+                  ...floatingStyles,
+                },
+              })}
+              className={cn(classes.itemsWrapper, classes[size])}
+            >
+              {/* Map our children, and add props if it is a ComboboxItem */}
+              {React.Children.map(filteredItems, (child, index) => {
+                if (
+                  React.isValidElement(child) &&
+                  child.type === ComboboxItem
+                ) {
+                  const props = {
+                    key: index,
+                    ref(node: HTMLElement | null) {
+                      listRef.current[index] = node;
```

Comment:
We are setting the `index` in `useCombobox` with `cloneElement` and another `cloneElement` for `key` with a different index here again, this has to be able to be simplified 🤔

Reply:
> This look better to you? 🗡️ 

![antonio-banderas-oh](https://github.com/digdir/designsystem/assets/1070981/8a5161e7-7a8a-49a0-859f-fd2c317190fc)